### ISSUE TITLE
Best Model for Eval.py 

### ISSUE THREAD

**Issue by mtkmark** [2017-fall-DL-training-program/ImageCaption#18]
We got several out-come after  training 

1. pkl files contains variable for each layers/matrix , so we got best value after training. I understand. 
2. for pth files, there are models description.  why do we have "best" models . is it just one models ? 

Thanks
Mark 

-rw-rw-r-- 1 student38 student38    289623 Nov 16 01:19 infos_1-best.pkl
-rw-rw-r-- 1 student38 student38 170494134 Nov 16 01:19 model-cnn-best.pth
-rw-rw-r-- 1 student38 student38  78777561 Nov 16 01:19 model-best.pth
-rw-rw-r-- 1 student38 student38    908614 Nov 16 01:19 histories_1.pkl
-rw-rw-r-- 1 student38 student38    289623 Nov 16 01:19 infos_1.pkl
-rw-rw-r-- 1 student38 student38 338273633 Nov 16 01:19 optimizer-cnn.pth
-rw-rw-r-- 1 student38 student38 157522139 Nov 16 01:18 optimizer.pth
-rw-rw-r-- 1 student38 student38 170494134 Nov 16 01:18 model-cnn.pth
-rw-rw-r-- 1 student38 student38  78777561 Nov 16 01:18 model.pth

**Comment by connie980149** [comment]
pkl files only save the training information like iteration, epoch, best score...etc.
pth file is saving the parameters of models.
The parameters will be modified during training, the "best" model saves the weights when the validation score is the best.

**Comment by fansia** [comment]
Hi Connie,

Could we read training information like iteration/epoch/training loss etc from pkl? and How?

Thanks,
Eric

**Comment by connie980149** [comment]
After loading the pkl file (ex. **infos = cPickle.load(f)**), the **infos** is a dictionary of python, you can use, for example, infos['iter'] to get the iteration saved in pkl file.
Also you can get training loss from histories['loss_history'] in histories.pkl.
